### ISSUE TITLE
Bug DevToolsDriver removes all cached elements when page loads new frames

### ISSUE THREAD

**Issue by kyryloonufriiev** [webdriverio/webdriverio#6724]
**Environment (please complete the following information):**
 - **WebdriverIO version:** 7.4.6
 - **Mode:** standalone
 - **Node.js version:** 12.20.2
 - **NPM version:** 6.14.11
 - **Browser name and version:** Chrome 89
 - **Platform name and version:** MacOS 11.2.3
 - **Additional wdio packages used (if applicable):** devtools

**Config of WebdriverIO**
`automationPrococol: 'devtools'`

**Describe the bug**
I have an issue with `devtools` package. It creates elements cache with fake ids and saves elements in an element store. But in order to clear unused references, a wrong strategy was chosen. The implementation listens to `framenavigated` event from page object and clear element store. But this event will be fired when any of the iframes on the page will be reloaded or a new frame will be added. In my opinion, this is wrong behavior, if you want to clear element references when the page itself is reloaded/navigated then before clearing (in `framenavigatedHandler` method) you should check that navigated frame (it will be the first argument of this method) is the main one (e.g. `!navigatedFrame.parentFrame()`). The superior solution will store elements together with the frame object reference, and remove elements only for the specific frame that was navigated.

**To Reproduce**
1. Find element
2. Run script like this
```js
browser.execute(() => document.head.append(document.createElement('iframe')))
```
3. Try to use the element found in step 1. (e.g. `browser.execute(el => el, el_from_step_1)`)

**Expected behavior**
The element should not be removed from the cache.

**Comment by christian-bromann** [comment]
Thanks for reporting @kyryloonufriiev . Any contributions would be appreciated.

**Comment by christian-bromann** [comment]
fixed in #6724
